frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditLayoutSetForSubform/EditLayoutSet/CreateNewSubformSection/CreateNewSubformSection.tsx

```diff
@@ -33,26 +38,33 @@ export const CreateNewSubformSection = ({
 }: CreateNewSubformSectionProps): React.ReactElement => {
   const { t } = useTranslation();
   const { validateLayoutSetName } = useValidateLayoutSetName();
-  const [nameError, setNameError] = useState<string>();
-  const [newDataModel, setNewDataModel] = useState<string>('');
+  const [newSubformNameError, setNewSubformNameError] = useState<string>();
   const [selectedDataModel, setSelectedDataModel] = useState<string>('');
   const [displayDataModelInput, setDisplayDataModelInput] = useState(false);
   const { createSubform, isPendingNewSubformMutation } = useCreateSubform();
+  const [isTextfieldEmpty, setIsTextfieldEmpty] = useState(true);
+
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: dataModelIds } = useAppMetadataModelIdsQuery(org, app, false);
+  const { data: appMetadata } = useAppMetadataQuery(org, app);
+  const dataTypeNames = extractDataTypeNamesFromAppMetadata(appMetadata);
+  const { validateName, nameError: dataModelNameError } = useValidateSchemaName(
+    dataModelIds,
+    dataTypeNames,
+  );
 
   const handleSubformName = (subformName: string) => {
     const subformNameValidation = validateLayoutSetName(subformName, layoutSets);
-    setNameError(subformNameValidation);
+    setNewSubformNameError(subformNameValidation);
   };
 
   const handleCloseButton = () => {
     if (displayDataModelInput) {
-      setNewDataModel('');
```

Comment:
Consider adding `setNewSubformNameError("")` here to clear the error message when closing the "create new data model" input field. Without this, I think the error message will persist if the user reopens new data model :)